### DISCUSSION TITLE
Why have or not the commit hash?

### DISCUSSION THREAD

**Discussion by cipri-tom** [MiroPsota/torch_packages_builder#1] [General]
Hi ! 

Thank you so much for building detectron2 for all those versions ! It's such a time saver !! 🥳 

This is a curiosity question. Why do you provide the option to have or not the commit hash ? When would it make sense to specify, and when to leave out?

Thank you !

**Comment by MiroPsota** [comment]
Glad to hear that it's helpful!

I use commit IDs to differentiate between commits with the same version number in a repo.
In the case of detectron2, there's the `v0.6` tag in the repository which corresponds to the officially released version 0.6 (Oct 26, 2021), marked as `0.6` without a commit hash. Since that's really old, I also built the newest commit available some time ago, marked as `0.6+864913f` (Jan 9, 2024). The part `+<something>` is a [local version identifier](https://peps.python.org/pep-0440/#local-version-identifiers), and I use it as the identifier of the commit that works with pip.

I should add it to the README.

**Comment by cipri-tom** [comment]
It makes a lot of sense ! 

Thanks for building the latest version ! It's the one most people are after, I believe. 

You really should write some stack overflow posts, and medium articles. This deserves a lot more visibility.
